src/Build/Construction/ProjectElementContainer.cs

```diff
@@ -64,32 +65,41 @@ internal ProjectElementContainer(XmlElement xmlElement, ProjectElementContainer
         /// </summary>
         public IEnumerable<ProjectElement> AllChildren => GetChildrenRecursively();
 
+        internal IEnumerable<T> GetAllChildrenOfType<T>()
+            where T : ProjectElement
+            => GetChildrenRecursivelyOfType<T>();
+
         /// <summary>
         /// Get enumerable over all the children
         /// </summary>
         public ICollection<ProjectElement> Children
         {
             [DebuggerStepThrough]
-            get
-            {
-                return new Collections.ReadOnlyCollection<ProjectElement>(
-                        new ProjectElementSiblingEnumerable(FirstChild));
-            }
+            get => new Collections.ReadOnlyCollection<ProjectElement>(new ProjectElementSiblingEnumerable(FirstChild));
         }
 
+        /// <summary>
+        /// Use this instead of <see cref="Children"/> to avoid boxing.
+        /// </summary>
+        internal ProjectElementSiblingEnumerable ChildrenEnumerable => new ProjectElementSiblingEnumerable(FirstChild);
```

Comment:
Hmm, interesting idea. It seems to work except I'm unsure how to handle this for UTs since it's a internal member and not all UTs have visibility. I _could_ InternalsVisibleTo, but not sure if that's desirable here or not (MSBuild team feel free to express an opinion)

Reply:
We can skip the ban enforcement in UTs. I'll push a commit with that shortly.